File: urbanairship-preference-center-react-native/android/src/main/java/com/urbanairship/reactnative/preferenceCenter/events/OpenPreferenceCenterEvent.java
package com.urbanairship.reactnative.preferenceCenter.events;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;
import com.urbanairship.reactnative.Event;

public class OpenPreferenceCenterEvent implements Event {

    private static final String OPEN_PREFERENCE_CENTER_EVENT = "com.urbanairship.open_preference_center";
    private final String preferenceCenterID;

    public OpenPreferenceCenterEvent(String preferenceCenterID) {
        this.preferenceCenterID = preferenceCenterID;
    }

    @NonNull
    @Override
    public String getName() {
        return OPEN_PREFERENCE_CENTER_EVENT;
    }

    @NonNull
    @Override
    public ReadableMap getBody() {
        WritableMap map = Arguments.createMap();

        if (preferenceCenterID != null) {
            map.putString("preferenceCenterID", preferenceCenterID);
        }
        return map;
    }

    @Override
    public boolean isForeground() {
        return false;
    }
}

File: urbanairship-preference-center-react-native/android/src/main/java/com/urbanairship/reactnative/preferenceCenter/events/OpenPreferenceCenterEvent.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.preferenceCenter.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap
import com.urbanairship.reactnative.Event

class OpenPreferenceCenterEvent(private val preferenceCenterId: String?) : Event {

    override val name: String
        get() = OPEN_PREFERENCE_CENTER_EVENT

    override val body: ReadableMap
        get() {
            val map = Arguments.createMap()
            preferenceCenterId?.let {
                map.putString("preferenceCenterId", preferenceCenterId)
            }
            return map
        }

    override val isForeground: Boolean
        get() = false

    companion object {

        private const val OPEN_PREFERENCE_CENTER_EVENT = "com.urbanairship.open_preference_center"

    }
}